### DISCUSSION TITLE
On-demand Revalidation From the Backend (CMS like starpi) - Why revalidation takes time ?

### DISCUSSION THREAD

**Discussion by saver711** [vercel/next.js#66483] [App Router]
I am using route to revalidate my posts like this
![image](https://github.com/vercel/next.js/assets/28202010/df10f423-13b4-4cf5-ad4c-9b4631eb1747)

------
i am revalidation from Strapi code whenever any post is updated like this
![image](https://github.com/vercel/next.js/assets/28202010/1895b7cb-7e43-4e57-b85f-ced529b55cbe)

-------
Why doesn't changes reflect immediately?
When i navigate back and forth between home and posts page i don't see the changes
it takes like 12 navigations or something

**Comment by AdamZajler** [comment]
Meaby it is because of cache? Try viewing it in incognito :thinking:
